src/status_im/ui/screens/chat/views.cljs

```diff
@@ -236,21 +242,54 @@
 
 (defonce messages-list-ref (atom nil))
 
+(def initial-limit 5)
+(def second-initial-limit-step 6)
+(def third-initial-limit-step 11)
+(def default-limit-step 20)
+(def initial-threshold 0.5)
+(def default-threshold 2)
+
+(defonce messages-limit (reagent/atom initial-limit))
+
+(defonce limit-enabled? (reagent/atom true))
+
+(defn increment-limit [lim]
+  (+ lim
+     (cond
```

Comment:
maybe better to use case here and put the numerical values directly

Reply:
how is it better?

Reply:
case is an order of magnitude faster, and that not even considering that you are doing some additions here in the conditions that are not sure can be optimize by the compiler

It's obvious that this is going to be more efficient no?

`(case lim
     5 11
     11 16
     (+ lim 20)`

Reply:
1. relying on magic numbers is going to be an issue, I prefer to avoid doing this. If values are co-dependent, that should be expressed in the code.
2. The difference in efficiency in negligible compering to the whole rendering thing as long as this operation is not going to be executed repeatedly, repl on slow android device https://gist.github.com/rasom/b6a154bef04894c3d5ecff5fca6fbfcf

Reply:
Eric, 10000000 times, really? :)

Reply:
This isn't magic numbers, they are only used once in the code, what's the point of having 4 extra symbols in the registry for something so simple. Anyway it's nitpicking, but I think encapsulating the increment-limit within one function, in a simple case statement, without extra references dangling around is neater here.

Reply:
A docstring explaining why you use theses steps would be better than renaming the numbers 5 11 16 and 20

Reply:
It's only magic numbers if there is no explanation

Reply:
I disagree, because numbers are co-dependent. And as optimization this change improves nothing.

Reply:
Would that work for you then?

```
(def ^:const first-step 5)
(def ^:const second-step 11) 
(def ^:const third-step 16)
(def ^:const step-increment 20)

(defn increment-limit [lim]
(case lim
     first-step second-step
     second-step third-step
     (+ lim step-increment)))
```

I would still like to understand why the increments are +6 then +4 then +20, looks like the real magic numbers here

Reply:
> I would still like to understand why the increments are +6 then +4 then +20, looks like the real magic numbers here

At first, it is 5, then +6, then +11. Not `+6 then +4 then +20`.

And... well that's true, they are magic numbers (but not in terms of programming), and as I'm 99% sure we are going to change these numbers I prefer to code the relations between them. 

The way how I "inferred" those numbers: on a slower device (samsung a500d, 720 x 1280 pixels, 16:9 ratio (~294 ppi density)) I opened `#status` and `#status-core` chats during several days, and almost always not more than 5 messages were shown initially. 11 is the max number of messages which fits on that screen. Both number are device specific, and probably can be derived more precisely in some better way, but 11 is kind of a safe number for almost all devices, chances that more than 11 messages will be shown on the initial screen in a regular chat are quite low at the moment.
<img width=250 src=https://user-images.githubusercontent.com/2364994/58072936-e39e4280-7ba9-11e9-9cbd-2e4e1d29d7e3.png />
<img width=250 src=https://user-images.githubusercontent.com/2364994/58072938-e5680600-7ba9-11e9-8066-f552c0ff4f9d.png />
<img width=250 src=https://user-images.githubusercontent.com/2364994/58072924-db460780-7ba9-11e9-9e3d-6f9707bb55a2.png />
<img width=250 src=https://user-images.githubusercontent.com/2364994/58072930-de40f800-7ba9-11e9-9dda-d7b2620214a6.png />